step5/index.html

```diff
@@ -116,15 +138,16 @@
 
     const attenuation = this.root.getElementById('attenuation').checked;
     const amount = +this.root.getElementById('amount').value;
-    const size = amount * .8;
+    const size = this.root.getElementById('size').value * amount;
+	const opacity = this.root.getElementById('opacity').value;
 
     for (let y = amount; y < canvas.height; y += amount * 2) {
       for (let x = amount; x < canvas.width; x += amount * 2) {
         const index = ((y * canvas.width) + x) * 4;
         const [r,g,b] = this.data.slice(index, index+3);
-        const color = `rgb(${r},${g},${b})`;
+        const color = `rgba(${r},${g},${b},${opacity})`;
 
-        const weight = Math.max(r + g + b) / 256;
+		const weight = 1 - ( this.data[ index ] / 255 );
```

Comment:
I find this line kind of weird, because it's going to only weight the pixel based on the red pixel data? (`data[index]` is R, `data[index+1]` is G, ...)—but if the effect looks good, sure.

Reply:
I think that's somehow getting the alpha channel - it's the code Ricardo used, and it gets the identical effect to your max(r,g,b) so I'm not 100% sure what's going on there.

Reply:
The alpha channel is `data[index+3]` (r=0, g=1, b=2, a=3). But virtually no images that get dragged on will have an alpha channel (e.g. jpegs don't have it). So this is just the red value.

I don't mind or care, just noting it.

Reply:
no you're right, that's what's happening. The weird thing is doing the max (r,g,b) leads to an identical result. (or, close enough I can't tell by looking) I'm gonna go back to the max() function because that makes more sense.